### PR TITLE
Migrate ui lib to native-base

### PR DESCRIPTION
**Pull request by MSzalowski**
# Description

Transit existing UI library to native-base.

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Go through the application on each platform.

# Screenshot(s)

ios

# Checklist:

- [x] Add correct label to your pull request
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/constants/asyncStorageKeys.test.ts [hunk 1]
```diff
@@ -1,16 +0,0 @@
-import { ASYNC_STORAGE_KEYS } from './asyncStorageKeys'
```
Comment (MateuszRostkowski) [inline src/constants/asyncStorageKeys.test.ts:1]: Why did you remove this test file?

File: src/hooks/forms/useSignInForm.ts [hunk 2]
```diff
@@ -9,8 +9,8 @@ import { useAuth } from '../useAuth'
 import { SignInFormValues } from '~types/authForms'
 
 const defaultValues: SignInFormValues = {
-  email: 'test@example.com',
-  password: '123456',
+  email: '',
+  password: '',
```
Comment (MateuszRostkowski) [inline src/hooks/forms/useSignInForm.ts:13]: This values were left for debbuging and easier testing. You can do something like this here:

```suggestion
  // TODO: Reset this values when building production app
  email: 'test@example.com',
  password: '123456',
```

File: src/index.tsx [hunk 3]
```diff
@@ -25,21 +26,21 @@ const queryClient = new QueryClient({})
 
 const App = (): JSX.Element => {
   return (
-    <SafeAreaProvider>
```
Comment (MateuszRostkowski) [inline src/index.tsx:28]: Why did you remove SafeAreaProvider?

File: src/index.tsx [hunk 4]
```diff
@@ -25,21 +26,21 @@ const queryClient = new QueryClient({})
...
+    <NativeBaseProvider theme={theme} colorModeManager={colorModeManager} config={nativeBaseConfig}>
       <NotificationsProvider>
-        <QueryClientProvider client={queryClient}>
-          <AuthProvider>
+        <AuthProvider>
```
Comment (MateuszRostkowski) [inline src/index.tsx:31]: When developing in the future AuthProvider might be using react query and it will not work when it's outside provider

File: src/providers/ColorSchemeProvider.tsx [hunk 5]
```diff
@@ -43,14 +45,15 @@ export const ColorSchemeProvider: FC = ({ children }) => {
     (newColorScheme: SettingColorSchemeName) => {
       const oldColorScheme = colorSchemeSetting
       setColorSchemeSetting(newColorScheme)
+      setColorMode(newColorScheme)
```
Comment (MateuszRostkowski) [inline src/providers/ColorSchemeProvider.tsx:48]: There can be some bugs after first installation of the app. When I'm analyzing videos that you provided looks like this can be issue with this.

Maybe you should run function `setColorMode` in `useEffect` function too?

File: src/screens/ApplicationInfoScreen.tsx [hunk 6]
```diff
@@ -1,18 +1,16 @@
...
 
 export const ApplicationInfoScreen = (): JSX.Element => {
   const { i18n } = useTranslation()
-  const { s } = useTheme()
   usePreventGoBack()
 
   return (
-    <ScrollView contentContainerStyle={[s.flex1, s.justifyStart, s.p4]}>
-      <Text>
+    <ScrollView p={4}>
```
Comment (MateuszRostkowski) [inline src/screens/ApplicationInfoScreen.tsx:12]: Before styles was added to content Container and it was because scroll indicator looks better for this specific case. Did you check if this is working fine with this setup?

File: src/screens/ComponentsScreen.tsx [hunk 7]
```diff
@@ -1,12 +1,14 @@
 import * as Linking from 'expo-linking'
-import { ScrollView } from 'react-native'
+import { VStack, Heading, Divider, Button, ScrollView } from 'native-base'
 
-import { Button, Text } from '~components'
-import { useCallback, useNotifications, useTheme } from '~hooks'
+import { useCallback, useNotifications, useTranslation } from '~hooks'
+
+const headingSizes = ['xs', 'sm', 'md', 'lg', '2xl', '3xl', '4xl']
+const buttonVariants = ['solid', 'outline', 'subtle', 'ghost']
```
Comment (MateuszRostkowski) [inline src/screens/ComponentsScreen.tsx:7]: You can add const assertion here and then you can remove 
```
{/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
{/* @ts-expect-error */}
```

```suggestion
const headingSizes = ['xs', 'sm', 'md', 'lg', '2xl', '3xl', '4xl'] as const
const buttonVariants = ['solid', 'outline', 'subtle', 'ghost'] as const
```

File: src/services/ColorSchemeManager.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import type { StorageManager, ColorMode } from 'native-base'
+
+export const colorModeManager: StorageManager = {
+  get: async () => {
+    try {
+      const val = await AsyncStorage.getItem('@color-mode')
```
Comment (MateuszRostkowski) [inline src/services/ColorSchemeManager.ts:7]: There is a file named asyncStorageKeys, add it there. Also change this value to '@theme/native-base-color-mode', because we have color sheme in our app and it can be hard to find difference between this two

**Review by MateuszRostkowski — CHANGES_REQUESTED** [review]
Great work 🎉 . 
 
Please add this Todo comments that you added in PR description to readme, you can add some detailed informations for the future :D

File: src/index.tsx (hunk 3 above)
Comment (MSzalowski) [reply src/index.tsx:28]: That provider isn't needed as `NativeBaseProvider` already wraps its children with `SafeAreaProvider`.

Maybe it would be good to leave a comment in a render tree informing that `SafeArea` features are available due to `NativeBase` integration?

Comment (MateuszRostkowski) [reply src/index.tsx:28]: It's good to know that! :D 

Yes, it's good idea to add some informations about SafeAreaProvider because it can be confusing for new developers

File: src/screens/ApplicationInfoScreen.tsx (hunk 6 above)
Comment (MSzalowski) [reply src/screens/ApplicationInfoScreen.tsx:12]: I've checked it and the indicator positions itself correctly. But I found another bug:

Indicator on iOS is visible only when the content of the ScrollView makes the ScrollView scrollable (ok, it is always scrollable but when there is more content than visible on a screen). So that - bug isn't visible now.

But - if there were more content, the indicator's color would not change according to the modal (dark indicator on the white screen and vice versa). I don't see any issue regarding this either on react-native or native-base repository so I am not sure why that is happening.

Nevertheless, we can fix that by changing `indicatorStyle` based on a set color scheme like this:
```tsx
    import { useColorMode } from 'native-base'

    
    const ApplicationInfo: FC = () => {
    const { colorMode } = useColorMode

    return (<ScrollView indicatorStyle={colorMode === 'dark' ? 'white' : 'black'}>
    ...
    </ScrollView>)
```

File: src/services/ColorSchemeManager.ts (hunk 8 above)
Comment (MSzalowski) [reply src/services/ColorSchemeManager.ts]: Imo we can use the same key for NB color scheme manager as they point to exact same value. They depends on each other

File: src/providers/ColorSchemeProvider.tsx (hunk 5 above)
Comment (MSzalowski) [reply src/providers/ColorSchemeProvider.tsx:49]: I made some little refactor that changed the initialization of a theme a bit. Unless I didn't make any mistake, executing `setColorMode` in `useEffect` is no longer necessary

File: src/screens/ApplicationInfoScreen.tsx (hunk 6 above)
Comment (MateuszRostkowski) [reply src/screens/ApplicationInfoScreen.tsx:12]: Okay that's good to know. We need to remember about this in the future, but it's not a big issue :P

File: src/components/Input.tsx [hunk 9]
```diff
@@ -1,129 +1,116 @@
...
-  inputContainerStyle?: StyleProp<ViewStyle>
+  helperText?: string
+  errorMessage?: string
+  errorIcon?: JSX.Element
   onInputBlur?: () => ChangeHandler
-  fullWidth?: boolean
-  type?: InputTypes
 }
 
+const layoutPropsKeys = [
```
Comment (kacgrzes) [inline src/components/Input.tsx:27]: Is it really necessary with native base?

File: src/contexts/ColorSchemeContext.ts [hunk 10]
```diff
@@ -6,7 +6,7 @@ export type ColorSchemeContextType =
   | {
       colorSchemeSetting: SettingColorSchemeName
       colorScheme: ColorSchemeName
-      setColorSchemeSetting: (newColorScheme: SettingColorSchemeName) => void
+      setColorSchemeSetting: (newColorScheme: ColorSchemeName) => void
```
Comment (kacgrzes) [inline src/contexts/ColorSchemeContext.ts:9]: I'm not sure if this file is needed. There is color mode in Native Base

File: src/providers/ColorSchemeProvider.tsx [hunk 11]
```diff
@@ -1,10 +1,11 @@
 import { useAsyncStorage } from '@react-native-async-storage/async-storage'
-import { FC, useEffect } from 'react'
+import { useColorMode } from 'native-base'
```
Comment (kacgrzes) [inline src/providers/ColorSchemeProvider.tsx:2]: We should not need this file as we have `colorModeManager` https://docs.nativebase.io/next/color-mode#h2-persisting-the-color-mode

File: src/screens/SignUpScreen.tsx [hunk 12]
```diff
@@ -1,78 +1,98 @@
...
           pattern: {
             value: REGEX.EMAIL,
             message: t('form.invalid_email_format'),
           },
         }}
-        containerStyle={[s.mB2]}
+        isRequired
+        mb={2}
       />
       <ControlledInput
```
Comment (kacgrzes) [inline src/screens/SignUpScreen.tsx:58]: I think that some of the props set on ControlledInput could be set as default inside of the component. What do you think?

File: src/screens/TypographyScreen.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+  const changeColorScheme = useCallback(
+    () => colorScheme.setColorSchemeSetting(colorScheme.colorScheme === 'dark' ? 'light' : 'dark'),
+    [colorScheme]
+  )
+
+  return (
+    <ScrollView>
+      <Center>
+        <Switch my={8} onChange={changeColorScheme} />
+        <Text fontSize="xs">xs</Text>
```
Comment (kacgrzes) [inline src/screens/TypographyScreen.tsx:17]: Can we create an array an map over it?

**Review by kacgrzes — COMMENTED** [review]
I left some small suggestions

File: src/screens/SignUpScreen.tsx (hunk 12 above)
Comment (MSzalowski) [reply src/screens/SignUpScreen.tsx:58]: I don't see any common props to make them set with some default value - any suggestions? 🤔

File: src/components/Input.tsx (hunk 9 above)
Comment (MSzalowski) [reply src/components/Input.tsx:27]: In the way, I've built that input - yes - layout propses would be passed to both FormControl and Input.

File: src/components/Atoms.tsx [hunk 14]
```diff
@@ -1,55 +1,23 @@
...
-import { View, ViewProps, TouchableOpacity, TouchableOpacityProps, ViewStyle } from 'react-native'
+import { Box, Pressable, IPressableProps, IBoxProps, StyledProps, Theme } from 'native-base'
+import { ViewStyle } from 'react-native'
 
-import { useTheme } from '~hooks'
-import { generateStyledSystem, SpacingValue, StyledProps } from '~utils'
-
-export type BoxProps = StyledProps & ViewProps
-export type TouchableProps = StyledProps & TouchableOpacityProps
 export type GradientProps = StyledProps & LinearGradientProps
```
Comment (MateuszRostkowski) [inline src/components/Atoms.tsx:5]: I think this will not work for gradient 🤔 

Or you should change the implementation of gradient. Imo you can remove StyledProps for now here
```suggestion
export type GradientProps = LinearGradientProps
```

Comment (MSzalowski) [reply src/components/Atoms.tsx:5]: Good catch, definitely it shouldn't be here

**Review by MateuszRostkowski — APPROVED** [review]

File: src/components/Checkbox.tsx [hunk 15]
```diff
@@ -1,73 +1,50 @@
...
-
-interface ControllerFieldProps {
-  field: ControllerRenderProps
 }
 
-export const Checkbox = ({
-  disabled,
-  value,
-  onChangeValue,
+export const Checkbox: React.FC<CheckboxProps> = ({
```
Comment (kacgrzes) [inline src/components/Checkbox.tsx:13]: I think this is more than a `Checkbox` as it's wrapped with `FormControl`. I imagine that there could be such case when we would like to just use Checkbox component without any wrappers, helper texts, errors. 

Maybe we should split it in such way:
Input, Checkbox, Switch / Toggle, Button, Radio, Select, Slider - raw, but already styled components

Then FormControl components with label, helperText, error. I'd named them Field

I would first create generic Field component and then wrap Checkbox and Input with Field, then I would do:

```ts
Field.Input
Field.Checkbox
Fielx.OtherComponents
```

Last, but not least, we could Create `FieldWithController` to wrap the whole field component with Contoller.

This is just an idea, but IMO, each abstraction should be on it's own, so we could:
- use regular input components without FormControl
- use FormControl without react-hook-form Controller
- use generic Controller components

Let's discuss this further @MSzalowski

File: src/components/Input.tsx [hunk 16]
```diff
@@ -1,129 +1,116 @@
...
+            inputProps.type === 'password' ? (
+              <Touchable mr={2} onPress={toggleSecurePassword}>
+                <Feather name={iconName} color={colors.gray['400']} size={24} />
+              </Touchable>
+            ) : (
+              inputProps.rightElement
+            )
+          }
+        />
+        {helperText && <FormControl.HelperText>{helperText}</FormControl.HelperText>}
```
Comment (kacgrzes) [inline src/components/Input.tsx:104]: `HelperText` could be wrapped with our own component just to apply styles, the same goes with `ErrorMessage`.

What I'd like us to do is to go with the most simple abstraction up to the more advanced use cases, so:

- basic atoms (Input, Checkbox, HelperText, ErrorMessage, etc
- molecule - Field, FieldController, etc,

File: src/screens/SettingsScreen.tsx [hunk 17]
```diff
@@ -1,30 +1,42 @@
...
+
   return (
-    <ScrollView contentContainerStyle={[s.flex1, s.itemsCenter, s.justifyCenter]}>
-      <Text>Current theme: {colorSchemeSetting}</Text>
-      {colorSchemesList.map((scheme) => {
-        const isSelected = scheme === colorSchemeSetting
+    <ScrollView>
+      <Center>
+        <Text fontSize="2xl" bold mb={2}>
+          Current theme: {colorMode}
```
Comment (kacgrzes) [inline src/screens/SettingsScreen.tsx:18]: move this text to i18n

File: src/screens/SettingsScreen.tsx [hunk 18]
```diff
@@ -1,30 +1,42 @@
...
+              onPress={handleColorSchemeSettingChange(scheme)}
+            >
+              {`${scheme}${isSelected ? ' - selected' : ''}`}
+            </Button>
+          )
+        })}
 
-      <Button style={[s.mT10]} onPress={signOut} title="Sign out!" />
+        <Button colorScheme="danger" mt={8} size="lg" width="64" onPress={signOut}>
+          Sign out!
```
Comment (kacgrzes) [inline src/screens/SettingsScreen.tsx:37]: This text should go to i18n

File: src/screens/ExamplesScreen.tsx [hunk 19]
```diff
@@ -22,23 +22,28 @@ export const ExamplesScreen = (props: ExamplesScreenProps): JSX.Element => {
...
+          {t('examples_screen.go_to_application_info')}
+        </Button>
+        <Button size="lg" width="64" mb={2} onPress={goToColors}>
+          {t('examples_screen.go_to_colors')}
+        </Button>
+        <Button size="lg" width="64" mb={2} onPress={goToComponents}>
+          {t('examples_screen.go_to_components')}
+        </Button>
+        <Button size="lg" width="64" mb={2} onPress={goToTypography}>
+          Go to Typography
```
Comment (kacgrzes) [inline src/screens/ExamplesScreen.tsx:38]: to i18n

File: src/screens/DetailsScreen.tsx [hunk 20]
```diff
@@ -43,10 +40,10 @@ export const DetailsScreen = (props: DetailsScreenProps): JSX.Element => {
         onChange={handleSheetChanges}
         backdropComponent={renderBackdrop}
       >
-        <View style={[s.p4]}>
+        <Box p={4}>
           <Text color="black">Awesome 🎉</Text>
```
Comment (kacgrzes) [inline src/screens/DetailsScreen.tsx:44]: To i18n

File: src/screens/DetailsScreen.tsx [hunk 21]
```diff
@@ -32,9 +29,9 @@ export const DetailsScreen = (props: DetailsScreenProps): JSX.Element => {
   )
 
   return (
-    <View style={[s.flex1, s.justifyCenter, s.itemsCenter]}>
+    <Center>
       <Text>This is details screen</Text>
-      <Button onPress={openModal} title="Open BottomSheetModal" style={[s.mB2]} />
+      <Button onPress={openModal}>Open BottomSheetModal</Button>
       <Text>Screen params {JSON.stringify(params)}</Text>
```
Comment (kacgrzes) [inline src/screens/DetailsScreen.tsx:35]: To i18n

File: src/screens/DetailsScreen.tsx [hunk 22]
```diff
@@ -32,9 +29,9 @@ export const DetailsScreen = (props: DetailsScreenProps): JSX.Element => {
   )
 
   return (
-    <View style={[s.flex1, s.justifyCenter, s.itemsCenter]}>
+    <Center>
       <Text>This is details screen</Text>
```
Comment (kacgrzes) [inline src/screens/DetailsScreen.tsx:33]: To i18n

File: src/screens/SignInScreen.test.tsx [hunk 23]
```diff
@@ -0,0 +1,56 @@
+import { SignInScreen } from './SignInScreen'
+
+import { act, cleanup, fireEvent, render } from '~utils/testUtils'
+
+afterEach(cleanup)
+
+describe('SignInScreen', () => {
+  it('should render SignIn screen with the key elements', () => {
+    const { getByPlaceholderText, getByRole, getAllByRole } = render(<SignInScreen />)
+    const emailInput = getByPlaceholderText('john@doe.com')
```
Comment (kacgrzes) [inline src/screens/SignInScreen.test.tsx:10]: I don't think that getting elements by placeholder text is good.

We should rather use `accessibilityRole`, '`accessibilitylabel`, `testID`

File: src/screens/SignInScreen.test.tsx [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+    const signInButton = getByTestId('signInButton')
+
+    expect(signInButton).toBeDefined()
+    fireEvent.changeText(emailInput, '')
+    fireEvent.changeText(passwordInput, '')
+    fireEvent.press(signInButton)
+
+    await act(async () => {
+      update(<SignInScreen />)
+      const requiredInputs = await findAllByText('This field is required')
```
Comment (kacgrzes) [inline src/screens/SignInScreen.test.tsx:34]: Same as above, when i18n provider is different that `en` then the test would fail

File: src/screens/SignInScreen.tsx [hunk 25]
```diff
@@ -1,76 +1,94 @@
...
       {/* TODO: Remove this after implementing signing in with backend  */}
-      <Box alignItems="center">
-        <Text.H5>{t('sign_in_screen.dont_have_an_account')}</Text.H5>
-        <Box mt={2} />
-        <Button onPress={goToSignUp} title="Sign up" variant="Flat" />
-      </Box>
-      <Box mt={8} />
-      <Text.BodyBold>Correct credentials</Text.BodyBold>
-      <Text.BodyRegular color="gray500" center>
+      <Text bold>Correct credentials</Text>
```
Comment (kacgrzes) [inline src/screens/SignInScreen.tsx:88]: Move to i18n

File: src/screens/TypographyScreen.tsx [hunk 26]
```diff
@@ -0,0 +1,54 @@
...
+        <Text fontSize="4xl">Text.fontSize: </Text>
+        {fontSizes.map((fontSize) => (
+          <Text key={fontSize} fontSize={fontSize}>
+            Text - {fontSize}
+          </Text>
+        ))}
+
+        <Divider />
+
+        <Text fontSize="4xl">Heding.size: </Text>
```
Comment (kacgrzes) [inline src/screens/TypographyScreen.tsx:45]: Move to i18n

**Review by kacgrzes — CHANGES_REQUESTED** [review]
I left some more suggestions and feedback @MSzalowski

File: src/screens/SignInScreen.tsx (hunk 25 above)
Comment (MSzalowski) [reply src/screens/SignInScreen.tsx:88]: I would suggest not moving that label to i18n as it seems to be temporary code, under the `TODO` section. Moving it to i18n would require from developer to keep in mind checking one more place in case of removal them.

**Review by MateuszRostkowski — APPROVED** [review]

File: src/components/molecules/Field.tsx [hunk 27]
```diff
@@ -0,0 +1,52 @@
+// import { FormControl } from 'native-base'
```
Comment (kacgrzes) [inline src/components/molecules/Field.tsx:1]: Lot's of comments here. Are we getting rid of that?

File: src/screens/SignInScreen.test.tsx (hunk 23 above)
Comment (kacgrzes) [reply src/screens/SignInScreen.test.tsx:10]: What about this one?

File: src/screens/SignInScreen.test.tsx (hunk 24 above)
Comment (kacgrzes) [reply src/screens/SignInScreen.test.tsx:34]: IMO: we shouldn't use text to find elements. Even in test examples

**Review by kacgrzes — CHANGES_REQUESTED** [review]

**Review by kacgrzes — APPROVED** [review]
The only thing that I found is that some of the tests still use `findAllByText`, but I don't think this is a big deal. @MSzalowski do you plan to commit something else to this branch? If not, then let's merge it
